src/dst/src/Distributed.tcl

```diff
@@ -69,8 +75,29 @@ proc run_load_balancer { args } {
   }
   dst::run_load_balancer $host $port
 }
-
+sta::define_cmd_args "add_worker_address" {
+    [-host host]
+    [-port port]
+    [-threads threads]
+}
 proc add_worker_address { args } {
-  sta::check_argc_eq2 "add_worker_address" $args
-  dst::add_worker_address [lindex $args 0] [lindex $args 1]
+  sta::parse_key_args "add_worker_address" args \
+    keys {-host -port -threads} \
+    flags {}
+  if { [info exists keys(-host)] } {
+    set host $keys(-host)
+  } else {
+    utl::error DST 16 "-host is required in add_worker_address cmd."
+  }
+  if { [info exists keys(-port)] } {
+    set port $keys(-port)
+  } else {
+    utl::error DST 17 "-port is required in add_worker_address cmd."
+  }
+  if { [info exists keys(-threads)] } {
+    set threads $keys(-threads)
+  } else {
+    set threads 10
+  }
```

Comment:
Future idea: use -1 to trigger a request to the worker to see how many CPUs it has.